src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +53,131 @@
       document.getDocumentElement().normalize();
+      return Optional.of(document);
     } catch (ParserConfigurationException | SAXException e) {
       // TODO: Alert the user.
       System.out.println("XML parsing error");
-      return;
+      return Optional.empty();
     }
-    NodeList nodeList = document.getElementsByTagName(TEXT_TAG);
+  }
+
+  /**
+   * Puts each transcript line from {@code document} in datastore as its own entity.
+   *
+   * @param request Indicates the lecture key to group the transcript lines under.
+   * @param document The XML file containing the transcript lines.
+   */
+  private void putTranscriptLinesInDatastore(long lectureId, Document document) {
+    NodeList nodeList = document.getElementsByTagName(TAG_TEXT);
     for (int nodeIndex = 0; nodeIndex < nodeList.getLength(); nodeIndex++) {
       Node node = nodeList.item(nodeIndex);
-      Element element = (Element) node;
-      String lineStart = element.getAttribute(START_ATTRIBUTE);
-      String lineDuration = element.getAttribute(DURATION_ATTRIBUTE);
-      String lineContent = node.getTextContent();
-      // TODO: Remove print statement. It is currently here for display purposes.
-      System.out.println(lineStart + " " + lineDuration + " "
-          + " " + lineContent);
+      datastore.put(createLineEntity(node, lectureId));
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    PreparedQuery preparedQuery = getLectureTranscriptQuery(lectureId);
+    ImmutableList<TranscriptLine> transcriptLines = getTranscriptLines(preparedQuery);
+    writeResponseAsJson(response, transcriptLines);
+  }
+
+  /**
+   * Returns the query for the lecture transcripts based on lecture id indicated in {@code request}.
+   */
+  private PreparedQuery getLectureTranscriptQuery(long lectureId) {
+    Key lectureKey = KeyFactory.createKey(PARAM_LECTURE, lectureId);
+    Filter lectureFilter =
+        new FilterPredicate(TranscriptLine.PROP_LECTURE, FilterOperator.EQUAL, lectureKey);
+
+    Query query = new Query(TranscriptLine.ENTITY_KIND)
+                      .setFilter(lectureFilter)
+                      .addSort(TranscriptLine.PROP_START, SortDirection.ASCENDING);
+    return datastore.prepare(query);
+  }
+
+  private ImmutableList<TranscriptLine> getTranscriptLines(PreparedQuery preparedQuery) {
+    ImmutableList.Builder<TranscriptLine> lineBuilder = new ImmutableList.Builder<>();
+    for (Entity entity : preparedQuery.asQueryResultIterable()) {
+      lineBuilder.add(TranscriptLine.fromLineEntity(entity));
     }
+    return lineBuilder.build();
+  }
+
+  /**
+   * Writes {@code list} as Json to {@code response}.
+   */
+  private void writeResponseAsJson(HttpServletResponse response, ImmutableList<TranscriptLine> list)
```

Comment:
please rename the variable 'list'. List is not a useful variable name. 

Please look at for best practices. https://docs.google.com/document/d/1_ldClEZLsI1cMpQl_BfeIlkjTsE2IiJ2PgvhABfO6vs/edit#heading=h.wx9q7rdncnvo

Reply:
It says: Avoid filler names, like temp, var, data, val, helper, util, etc. that don't aid the reader's understanding.

A better name would be transcriptLines

Reply:
Thanks for catching that! I think I originally wanted it vague to generalize the function but looking over it again, I realize that can't be possible since the list stores transcriptLine objects.